src/Proto.Actor/LocalContext.cs

```diff
@@ -525,5 +528,16 @@ private void ReceiveTimeoutCallback(object state)
         {
             Self.Request(Proto.ReceiveTimeout.Instance, null);
         }
+
+        public void ReenterAfter<T>(Task<T> target, Func<Task<T>,Task> action)

```

Comment:
It is the same as we introduced in Go which was called AwaitFuture there, it creates a continuation and runs the continuation within the actor concurrency constraint when the task has completed.
This allows the actor to process other messages while the task is running